sys/netinet/ip_output.c

```diff
@@ -198,7 +198,8 @@ ip_output_pfil(struct mbuf **mp, struct ifnet *ifp, int flags,
 	/* Or forward to some other address? */
 	if ((m->m_flags & M_IP_NEXTHOP) &&
 	    ((fwd_tag = m_tag_find(m, PACKET_TAG_IPFORWARD, NULL)) != NULL)) {
-		bcopy((fwd_tag+1), dst, sizeof(struct sockaddr_in));
+		bcopy((struct sockaddr_in *)(fwd_tag+1), dst,
```

Comment:
Was this needed to silence a warning?

Reply:
Yes.  The compiler assumed fwd_tag+1 was a `struct mtag` which contains a pointer and that `dst` wasn't suitably aligned.  However, I suspect that this could also break with subobject bounds if sizeof(struct mtag) is < sizeof(struct sockaddr_in).  I do wonder if we subobject bounds it would be possible to have a warning for memcpy/bcopy/memset if the passed in size is larger than the bounded size that will be set to help with these cases from the subobject bounds view.

Reply:
The warning might be helpful, I'm not sure if we would get too many warnings for memcpy/bcopy/memset when the src/dst is a `char *` or a similar generic buffer, perhaps we could warn only for composite data types?

Reply:
Oh, that's true, you'd have to ignore it for pointers to `void` and `char`, but even things like int arrays would perhaps be bogus, so perhaps too hard to make it not overly noisy.